attrd/commands.c

```diff
@@ -713,15 +752,18 @@ attrd_peer_update(crm_node_t *peer, xmlNode *xml, const char *host, bool filter)
         free_xml(sync);
 
     } else if(safe_str_neq(v->current, value)) {
-        crm_info("Setting %s[%s]: %s -> %s from %s", attr, host, v->current, value, peer->uname);
-        free(v->current);
-        if(value) {
-            v->current = strdup(value);
+        if (safe_str_eq(op, ATTRD_OP_UPDATE) || safe_str_eq(op, ATTRD_OP_UPDATE_BOTH)) {
+            crm_info("Setting %s[%s]: %s -> %s from %s", attr, host, v->current, value, peer->uname);
+            free(v->current);
+            if(value) {
+                v->current = strdup(value);
+            } else {
+                v->current = NULL;
+            }
+            changed = TRUE;
         } else {
-            v->current = NULL;
+             crm_trace("Unchanged %s[%s] from %s.(ATTRD_OP_UPDATE_DELAY)", attr, host, peer->uname);
```

Comment:
Hi Hideo,

This block of changes is unnecessary, because we've already returned before this point if op is ATTRD_OP_UPDATE_DELAY.

Everything else looks great, thanks.